core/primitives/src/transaction.rs

```diff
@@ -291,6 +292,38 @@ impl Into<transaction_proto::DeleteKeyTransaction> for DeleteKeyTransaction {
     }
 }
 
+#[derive(Hash, Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+pub struct AddBlsKeyTransaction {
+    pub nonce: u64,
+    pub originator: AccountId,
+    pub new_key: Vec<u8>,
+    pub proof_of_possession: Vec<u8>,
+}
+
+impl From<transaction_proto::AddBlsKeyTransaction> for AddBlsKeyTransaction {
+    fn from(t: transaction_proto::AddBlsKeyTransaction) -> Self {
+        AddBlsKeyTransaction {
+            nonce: t.nonce,
+            originator: t.originator,
+            new_key: t.new_key,
+            proof_of_possession: t.proof_of_possession,
+        }
+    }
+}
+
+impl Into<transaction_proto::AddBlsKeyTransaction> for AddBlsKeyTransaction {
+    fn into(self) -> transaction_proto::AddBlsKeyTransaction {
+        transaction_proto::AddBlsKeyTransaction {
+            nonce: self.nonce,
+            originator: self.originator,
+            new_key: self.new_key,
+            proof_of_possession: self.proof_of_possession,
+            unknown_fields: Default::default(),
+            cached_size: Default::default(),
+        }
+    }
+}
+
```

Comment:
I added pretty logs in #620 
Please remove derive Debug add fmt::Debug implementation here:
```
impl fmt::Debug for AddBlsKeyTransaction {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        f.debug_struct("AddBlsKeyTransaction")
            .field("nonce", &format_args!("{}", &self.nonce))
            .field("originator", &format_args!("{}", &self.originator))
            .field("new_key", &format_args!("{}", logging::pretty_utf8(&self.new_key)))
            .field("proof_of_possession", &format_args!("{}", logging::pretty_utf8(&self.proof_of_possession)))
            .finish()
    }
}
```